packages/template-retail-react-app/app/pages/product-list/index.jsx

```diff
@@ -195,25 +195,34 @@ const ProductList = (props) => {
         // If we aren't allowing for multiple selections, simply clear any value set for the
         // attribute, and apply a new one if required.
         if (!allowMultiple) {
+            const previousValue = searchParamsCopy.refine[attributeId]
             delete searchParamsCopy.refine[attributeId]
 
-            if (!selected) {
+            // Note the loose comparison, for "string != number" checks.
+            if (!selected && value.value != previousValue) {
                 searchParamsCopy.refine[attributeId] = value.value
             }
         } else {
             // Get the attibute value as an array.
             let attributeValue = searchParamsCopy.refine[attributeId] || []
-            let values = Array.isArray(attributeValue) ? attributeValue : attributeValue.split('|')
+
+            // Ensure that the value is still converted into an array if it's a `string` or `number`.
+            if (typeof attributeValue === 'string') {
+                attributeValue = attributeValue.split('|')
+            } else if (typeof attributeValue === 'number') {
+                attributeValue = [attributeValue]
+            }
```

Comment:
Originally the number values in the query string were not being parsed as numbers and were being treated as strings. Looks like we broke this logic after making the change to parse numbers. 

Thanks for catching that. 👍